build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -0,0 +1,180 @@
+                                           GenerateNuGetPackagesArchive">
+  </Target>
+
+  <Target Name="SetupStage">
+    <ItemGroup>
+      <CompileStageInputs Include="$(RepoRoot)/src/**/*.cs" Exclude="$(RepoRoot)/src/**/obj/**/*.cs" />
+      <CompileStageInputs Include="$(RepoRoot)/src/**/project.json;$(RepoRoot)/src/**/project.lock.json" />
+
+      <Stage Include="Stage1">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage1Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage1SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(Stage0Path)/dotnet.exe</DotnetExe>
+      </Stage>
+      <Stage Include="Stage2">
+        <Inputs>@(CompileStageInputs)</Inputs>
+        <StageDirectory>$(Stage2Directory)</StageDirectory>
+        <StageSymbolsDirectory>$(Stage2SymbolsDirectory)</StageSymbolsDirectory>
+        <DotnetExe>$(DotnetStage1)</DotnetExe>
+      </Stage>          
+    </ItemGroup>        
+  </Target>    
+
+  <Target Name="CompileStage"
+          Inputs="%(Stage.Identity);%(Stage.Inputs)"
+          Outputs="%(StageDirectory)/sdk/$(NugetVersion)/dotnet.dll">
+      <PropertyGroup>
+        <SrcDirectory>$(RepoRoot)/src</SrcDirectory>
+
+        <ExeSuffix Condition=" '$(OSName)' == 'win' ">.exe</ExeSuffix>
+        <ExeSuffix Condition=" '$(OSName)' != 'win' "></ExeSuffix>
+
+        <DynamicLibPrefix Condition=" '$(OSName)' == 'win' "></DynamicLibPrefix>
+        <DynamicLibPrefix Condition=" '$(OSName)' != 'win' ">lib</DynamicLibPrefix>
+
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'win' ">.dll</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(OSName)' == 'osx' ">.dylib</DynamicLibSuffix>
+        <DynamicLibSuffix Condition=" '$(DynamicLibSuffix)' == '' ">.so</DynamicLibSuffix>
+
+        <DotnetHostBaseName>dotnet$(ExeSuffix)</DotnetHostBaseName>
+        <DotnetHostFxrBaseName>$(DynamicLibPrefix)hostfxr$(DynamicLibSuffix)</DotnetHostFxrBaseName>
+        <HostPolicyBaseName>$(DynamicLibPrefix)hostpolicy$(DynamicLibSuffix)</HostPolicyBaseName>
+
+        <StageDirectory>%(Stage.StageDirectory)</StageDirectory>
+        <StageSymbolsDirectory>%(Stage.StageSymbolsDirectory)</StageSymbolsDirectory>
+        <SdkOutputDirectory>$(StageDirectory)/sdk/$(NugetVersion)</SdkOutputDirectory>            
+        <BinaryToCorehostifyRelDir>runtimes/any/native</BinaryToCorehostifyRelDir>
+        <BinaryToCorehostifyOutDir>$(SdkOutputDirectory)/$(BinaryToCorehostifyRelDir)</BinaryToCorehostifyOutDir>
+        <MSBuildTargetsDirectory>$(SdkOutputDirectory)/runtimes/any/native</MSBuildTargetsDirectory>
+      </PropertyGroup>
+
+      <ItemGroup>
+        <BinObj Include="$(SrcDirectory)/**/bin/**/*" />
+        <BinObj Include="$(SrcDirectory)/**/obj/**/*" />
+        <FilesToRemoveFromStageDirectory Include="$(StageDirectory)/**/*" />
+
+        <SharedFramework Include="$(IntermediateDirectory)/sharedFrameworkPublish/**/*.*" />
```

Comment:
Should this be `$(IntermediateDirectory)/sharedFrameworkPublish/**/*`, will `*.*` bring over extensionless binaries on non-windows?